Behavior of non-versioned and versioned cluster tasks (#45917)
openshift/openshift-docs: modules/op-difference-between-non-versioned-and-versioned-cluster-tasks.adoc

// This module is part of the following assembly:
//
// *cicd/pipelines/managing-nonversioned-and-versioned-cluster-tasks.adoc
:_content-type: CONCEPT
[id="difference-between-non-versioned-and-versioned-cluster-tasks_{context}"]

> = Difference between non-versioned and versioned cluster tasks

Reviewer: ```suggestion
= Differences between non-versioned and versioned cluster tasks
```

> 

Reviewer: - All modules and assemblies require a summary phrase that functions as an abstract. 
- Have you considered moving this table to modules/op-advantages-and-disadvantages-of-non-versioned-and-versioned-cluster-tasks.adoc ?

> .Difference between non-versioned and versioned cluster tasks

Reviewer: ```suggestion
.Differences between non-versioned and versioned cluster tasks
```

[options="header"]
|===

> | | Non-versioned cluster task | Versioned cluster task

Reviewer 2: shall we add a third row to this table `intent` (stable pipelines vs uptodate pipelines) even though it is mentioned in detail in advantages vs dis-advantages table.

Author: We are strongly discouraged about repeating things. If I have to mention intent here, I will have to take out those points from the "advantages and disadvantages" section. I would prefer to retain them where they are currently.

| Nomenclature  
| The non-versioned cluster task variant only contains the name of the cluster task. For example, the the name of the non-versioned variant of the Buildah cluster task installed with Operator v1.7 is `buildah`.  

> | The versioned cluster task variant contains the name of the cluster task, followed by the version as a suffix. For example, the the name of the versioned variant of the Buildah cluster task installed with Operator v1.7 is `buildah-v-1-7`.  

Reviewer: ```suggestion
| The versioned cluster task variant contains the name of the cluster task, followed by the version as a suffix. For example, the name of the versioned variant of the Buildah cluster task installed with Operator v1.7 is `buildah-v-1-7`.  
```

| Upgrade 

> | When you upgrade the Operator, the name of the non-versioned cluster task remains same. The updates for the new versions of cluster tasks are applied on top of the existing version. 

Reviewer: ```suggestion
| When you upgrade the Operator, the name of the non-versioned cluster task remains the same. The updates for the new versions of cluster tasks are applied on top of the existing version. 
```

> | When you upgrade the Operator, the new version of the cluster tasks are installed and the old versions are retained. By default, the latest compatible version of a versioned cluster task corresponds to the upgraded Operator. 

Reviewer 2: ```suggestion
| When you upgrade the Operator, the new version of the cluster tasks (eg: buildah-v-1-7) are installed. The old versions (eg: buildah-v-1-6) are retained without upgrades. By default, the latest compatible version of a versioned cluster task corresponds to the upgraded Operator. 
```

// This module is part of the following assembly:
//
// *cicd/pipelines/managing-nonversioned-and-versioned-cluster-tasks.adoc
:_content-type: CONCEPT
[id="difference-between-non-versioned-and-versioned-cluster-tasks_{context}"]

> = Difference between non-versioned and versioned cluster tasks

Reviewer: This title is incomplete. Please make the first word plural or use a definite article.
 ```suggestion
= Differences between non-versioned and versioned cluster tasks
```
or 
```suggestion
= The difference between non-versioned and versioned cluster tasks
```
